simple/msg_sockets.c

```diff
@@ -273,13 +288,42 @@ static int client_connect(void)
 	if (ret)
 		return ret;
 
+	switch (opts.dest_mode) {
+	case FT_DEST_ADDR_ONLY_IN_FI_CONN:
+		ret = fi_send(ep, sendBuf, strlen(sendBuf) + 1, fi_mr_desc(mr),
+					remote_fi_addr, NULL);
+		if (ret >= 0) {
+			FT_ERR("Unexpected result of fi_send - %d (ep %p)", ret, ep);
+			return -FI_EOTHER;
+		}
```

Comment:
I think this is the only piece of new code that's actually needed.  You're just wanting to see what happens if fi_send() is called prior to calling connect.  Since this is a coding error in the application, most providers are not going to want the overhead of checking the connection on every send call.  So an assertion in the provider is better than a run time failure, which would prevent fabtests from actually trying to catch this error.